pype/plugins/global/publish/extract_review.py

```diff
@@ -13,22 +19,1064 @@ class ExtractReview(pyblish.api.InstancePlugin):
+            ).format(**profile_filter_data))
+            return
+
+        self.log.debug("Matching profile: \"{}\"".format(json.dumps(profile)))
+
+        instance_families = self.families_from_instance(instance)
+        _profile_outputs = self.filter_outputs_by_families(
+            profile, instance_families
+        )
+        if not _profile_outputs:
+            self.log.info((
+                "Skipped instance. All output definitions from selected"
+                " profile does not match to instance families. \"{}\""
+            ).format(str(instance_families)))
+            return
+
+        # Store `filename_suffix` to save arguments
+        profile_outputs = []
+        for filename_suffix, definition in _profile_outputs.items():
+            definition["filename_suffix"] = filename_suffix
+            profile_outputs.append(definition)
+
+        # Loop through representations
+        for repre in tuple(instance.data["representations"]):
+            tags = repre.get("tags") or []
+            if (
+                "review" not in tags
+                or "multipartExr" in tags
+                or "thumbnail" in tags
+            ):
+                continue
+
+            input_ext = repre["ext"]
+            if input_ext.startswith("."):
+                input_ext = input_ext[1:]
+
+            if input_ext not in self.supported_exts:
+                self.log.info(
+                    "Representation has unsupported extension \"{}\"".format(
+                        input_ext
+                    )
+                )
+                continue
+
+            # Filter output definition by representation tags (optional)
+            outputs = self.filter_outputs_by_tags(profile_outputs, tags)
+            if not outputs:
+                self.log.info((
+                    "Skipped representation. All output definitions from"
+                    " selected profile does not match to representation's"
+                    " tags. \"{}\""
+                ).format(str(tags)))
+                continue
+
+            for output_def in outputs:
+                # Make sure output definition has "tags" key
+                if "tags" not in output_def:
+                    output_def["tags"] = []
+
+                # Create copy of representation
+                new_repre = copy.deepcopy(repre)
+
+                # Remove "delete" tag from new repre if there is
+                if "delete" in new_repre["tags"]:
+                    new_repre["tags"].remove("delete")
+
+                # Add additional tags from output definition to representation
+                for tag in output_def["tags"]:
+                    if tag not in new_repre["tags"]:
+                        new_repre["tags"].append(tag)
+
+                self.log.debug(
+                    "New representation tags: `{}`".format(new_repre["tags"])
+                )
+
+                # # QUESTION Why the hell we were adding tags to families?
```

Comment:
Who knows why we did this?

Reply:
no idea to be honest. I"m afraid it might be rubish from the first implementation. Sheck if it's used anywhere and get rid of it.